lib/Parse/ParseStmt.cpp

```diff
@@ -899,9 +899,6 @@ ParserResult<Stmt> Parser::parseStmtThrow(SourceLoc tryLoc) {
 
   ParserResult<Expr> Result = parseExpr(diag::expected_expr_throw);
 
-  if (Result.hasCodeCompletion())
-    return makeParserCodeCompletionResult<Stmt>();
-
```

Comment:
We need to propagate `hasCodeCompletion()`ness in `if (tryLoc.isValid() && exprLoc.isValid())` branch as well.
 how about something like:
```cpp
  bool hasCodeCompletion = Result.hasCodeCompletion();

  if (Result.isNull())
    Result = ...

  if (tryLoc.isValid() && exprLoc.isValid()) {
    Result = ...
  }

  if (hasCodeCompletion)
    Resullt.setHasCodeCompletion()
```

Reply:
This affects something like this invalid code:
```swift
try throw something().<HERE>
```

Reply:
Good catch! Updated it as suggested and added that as a test case. The code change doesn't seem to affect the completions produced in that case, but I figured it's probably still good to have the tests include it for code coverage.

Reply:
For completions inside function bodies, parser doesn't rely on `hasCodeCompletion()`. So it shouldn't be affected.

I believe 
```
do {
  try throw something().#HERE#
} catch _ {}
```
at top-level shows the effect.

Reply:
Thanks - let me update the test case.

Reply:
done!